cmd/segment-reaper/delete.go

```diff
@@ -53,9 +65,91 @@ func cmdDelete(cmd *cobra.Command, args []string) (err error) {
 		err = errs.Combine(err, inputFile.Close())
 	}()
 
-	_ = csv.NewReader(inputFile)
+	csvReader := csv.NewReader(inputFile)
+	csvReader.FieldsPerRecord = 5
+	csvReader.ReuseRecord = true
+
+	segmentsDeleted := 0
+	segmentsErrored := 0
+	segmentsSkipped := 0
+	for {
+		record, err := csvReader.Read()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			log.Error("error while reading record", zap.Error(err))
+			continue
+		}
+
+		projectID := record[0]
+		segmentIndex := record[1]
+		bucketName := record[2]
+		encodedPath := record[3]
+		creationDateFromReport, err := time.Parse(time.RFC3339Nano, record[4])
+		if err != nil {
+			log.Error("error while parsing date", zap.Error(err))
+			continue
+		}
+
+		encryptedPath, err := base64.StdEncoding.DecodeString(encodedPath)
+		if err != nil {
+			log.Error("error while decoding encrypted path", zap.Error(err))
+			continue
+		}
+
+		path := storj.JoinPaths(projectID, segmentIndex, bucketName, string(encryptedPath))
+		rawPath := storj.JoinPaths(projectID, segmentIndex, bucketName, encodedPath)
+
+		err = deleteSegment(ctx, db, path, creationDateFromReport, deleteCfg.DryRun)
+		if err != nil {
+			if errKnown.Has(err) {
+				segmentsSkipped++
+			} else {
+				segmentsErrored++
+			}
+			log.Error("error while deleting segment", zap.String("path", rawPath), zap.Error(err))
+			continue
+		}
+
+		log.Debug("segment deleted", zap.String("path", rawPath))
+		segmentsDeleted++
+	}
+
+	log.Info("summary", zap.Int("deleted", segmentsDeleted), zap.Int("skipped", segmentsSkipped), zap.Int("errored", segmentsSkipped))
+
+	return nil
+}
+
+func deleteSegment(ctx context.Context, db metainfo.PointerDB, path string, creationDate time.Time, dryRun bool) error {
+	pointerBytes, err := db.Get(ctx, []byte(path))
+	if err != nil {
+		return err
```

Comment:
I'm wondering if in case the pointer is not found it should count as skipped rather than as an erroned, so basically returns `errKnown.New(...` when `err == storage.ErrKeyNotFound`

Reply:
hmm I have no strong opinion here, I can change it you feel it will help.

Reply:
Let's wait for a thrid opinion.

Reply:
Yeah, it could return `errKnown.New("segment already deleted by user")` for consistency.